### PR TITLE
Feat: ARMv7 support

### PR DESCRIPTION
**Pull request by timfish**
- Allows libffi-rs to build for Raspberry Pi and other ARMv7 targets that don't have `longdouble`
- Fixes a failing test on Windows

### CODE REVIEW

**Comment by tov** [comment]
Thanks for the PR! This seems like a good thing to fix, but I wonder if there’s a simple way we could make it detect  which types are available rather than having to specify the ABIs explicitly in the source…

**Comment by timfish** [comment]
> I wonder if there’s a simple way we could make it detect which types are available

I'm open to suggestions but this is the only way I can think of!

**Comment by timfish** [comment]
It would be great to find a way to get something like this merged. I would like to publish a crate which depends on `libffi` but it won't build on Raspberry Pi and that's a platform we want to target.

I think it's safe to reduce the conditional compilation down to:
```rust
 #[cfg(not(target_arch = "arm"))]
```

The only other way I can think to do this is to parse the code generation output of `libffi-sys` and set a feature flag. It seems like this could be a lot more fragile that simply disabling `longdouble` support for arm. What other platforms don't support `longdouble`? Do we even have access to any of those platforms to test?

**Comment by YorickPeterse** [comment]
@timfish Now that 1.0.0 is released, could you rebase this MR and adjust if needed? I'd love to have these changes merged in :smile:

File: src/low.rs [hunk 1]
```diff
@@ -130,18 +130,23 @@ pub use raw::{ffi_abi, ffi_abi_FFI_DEFAULT_ABI, ffi_cif, ffi_closure, ffi_status
...
     pub use crate::raw::{
-        ffi_type_double as double, ffi_type_float as float, ffi_type_longdouble as longdouble,
-        ffi_type_pointer as pointer, ffi_type_sint16 as sint16, ffi_type_sint32 as sint32,
-        ffi_type_sint64 as sint64, ffi_type_sint8 as sint8, ffi_type_uint16 as uint16,
-        ffi_type_uint32 as uint32, ffi_type_uint64 as uint64, ffi_type_uint8 as uint8,
-        ffi_type_void as void,
+        ffi_type_double as double, ffi_type_float as float, ffi_type_pointer as pointer,
+        ffi_type_sint16 as sint16, ffi_type_sint32 as sint32, ffi_type_sint64 as sint64,
+        ffi_type_sint8 as sint8, ffi_type_uint16 as uint16, ffi_type_uint32 as uint32,
+        ffi_type_uint64 as uint64, ffi_type_uint8 as uint8, ffi_type_void as void,
```
Comment (YorickPeterse) [inline src/low.rs:136]: @timfish could you format this code with rustfmt if not done so already? I think this would make this block a bit easier to read.

Comment (timfish) [reply src/low.rs:136]: I think this was actually run through rust fmt but I'll double check!

Comment (timfish) [reply src/low.rs:136]: Yes, this is the output from rustfmt. I even tried it with nightly and get the same result!

Comment (YorickPeterse) [reply src/low.rs:136]: @timfish Ah perfect, thanks for clarifying.

**Comment by YorickPeterse** [comment]
@timfish Thanks!

**Pull request merged**